### PR TITLE
Marketplace: Add empty page for plans

### PR DESCRIPTION
**Pull request by candy02058912**
#### Proposed Changes

* Adds a new /plugins/plans/:site route with empty content.

#### Testing Instructions

* Visit http://calypso.localhost:3000/plugins/plans/testing705633960.wordpress\com with your own test site in the url
* View the new (empty) plans page:
Screenshot 2022-09-13 at 15-18-46 Plugins Plan Upgrade ‹ Business - Simple Site — WordPress com
* Confirm plugin details pages still load as this shares a similar route pattern

Related to #67574

### CODE REVIEW

File: client/my-sites/plugins/plans/index.jsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+import PluginsNavigationHeader from '../plugins-navigation-header';
+
+const Plans = () => {
+	const { referenceRef: navigationHeaderRef } = useScrollAboveElement();
+
+	const translate = useTranslate();
+
+	return (
+		<MainComponent wideLayout>
+			<PageViewTracker path={ '/plugins/plans' } title={ translate( 'Plugins > Plan Upgrade' ) } />
```
Comment (lsl) [inline client/my-sites/plugins/plans/index.jsx]:
```suggestion
			<PageViewTracker path="/plugins/plans" title="Plugins > Plan Upgrade" />
```
Not sure if we should be translating these - is there prior art where this is done? (I'm assuming the title ends up in tracks, we probably want the english version for comparing results using the title - if we do that)

File: client/my-sites/plugins/plans/index.jsx [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+			<DocumentHead title={ translate( 'Plugins > Plan Upgrade' ) } />
+
+			<PluginsNavigationHeader
+				navigationHeaderRef={ navigationHeaderRef }
+				categoryName={ translate( 'Plan Upgrade' ) }
+				category={ 'something needs to be here so that the top can show' }
+			/>
```
Comment (lsl) [inline client/my-sites/plugins/plans/index.jsx]:
```suggestion
```

Lets remove for now, I'd rather fix it than work around the problem but I think for now we can just leave it for a follow up to do the fix.

**Review by lsl — APPROVED** [review]

File: client/my-sites/plugins/index.web.js [hunk 3]
```diff
@@ -89,6 +89,18 @@ export default function ( router ) {
 		clientRender
 	);
 
+	router(
+		`/${ langParam }/plugins/plans`,
```
Comment (lsl) [inline client/my-sites/plugins/index.web.js]:
```suggestion
		`/${ langParam }/plugins/plans/:site`,
```

Lets scope this to site only. There should be a site selector middleware we can use too.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7525248

Thank you @candy02058912 for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7525248).

**Pull request merged**